add more documentation (#7)
mattermost/mattermost-operator: README.md

> # Mattermost Operator for Kubernetes ![CircleCI branch](https://img.shields.io/circleci/project/github/mattermost/mattermost-operator/master.svg) ![Alpha](https://img.shields.io/badge/alfa-in%20progress-yellow.svg) [![Community Server](https://img.shields.io/badge/Mattermost_Community-cloud_channel-blue.svg)](https://community.mattermost.com/core/channels/cloud)

Reviewer: Can we also wire up the images to link to CircleCI and whatever alfa-in is?

Author: removed

> Mattermost Operator for Kubernetes is under construction.

Reviewer: Can we merge this sentence and the `Status` section below? Seems redundant to have both.

Mattermost is a scalale, open source collaboration tool. It's written in Golang and React.

> The idea of this project is to build the Kubernetes Operator for Mattermost to mak more easy for Kubernetes users deploy and manage your Mattermost instance.

Reviewer: ```suggestion
This project offers a Kubernetes Operator for Mattermost to simplify deploying and managing your Mattermost instance.
```

## Getting Started

### Prerequisites

- Kubernetes cluster 1.8.0+.

> - `kubectl` configured to refer to relevant Kubernetes cluster.

Reviewer: Worth linking out to https://kubernetes.io/docs/reference/kubectl/overview/ here?
```suggestion
- `kubectl` configured for the relevant Kubernetes cluster. (https://kubernetes.io/docs/reference/kubectl/overview/)
```

> ### Create Operator and related resources

Reviewer: ```suggestion
### Install Operators
```

> To start Mattermost-Operator, we need to install the dependencies first.

Reviewer: Suggest eliminating this line:
```suggestion
```

and removing the `MySQL-Operator` and `Mattermost-Operator` headers below, to something like:
```
Install the mysql-operator for use by the mattermost-operator:
<kubectl commands>

Install the mattermost-operator:
<kubectl commands>
```

The fact that these manifests are in the `docs` folders does not -- IMO -- need to be called out explicitly, given they are in the commands being executed.

#### MySQL-Operator
To install MySQL-Operator apply the manifests that you can find in the `docs` folder

```
$ kubectl create ns mysql-operator

> $ kubectl apply -n mysql-operator -f https://github.com/mattertmost/mattermost-operator/blob/master/docs/mysql-operator/mysql-operator.yaml?raw=true

Reviewer: This URL doesn't resolve for me -- first, I note a typo `mattertmost`, but even then I can't find the file.

Reviewer 2: Yeah, I think those files are currently located in https://github.com/mattermost/mattermost-operator/tree/master/test/mysql

From my work making the provisioning server deploy the mattermost operator, I have a todo item that includes getting the K8s yaml resources in one spot, so maybe we should do that here with the documentation?

```

#### Mattermost-Operator
After the dependencies installed we need to deploy Mattermost-operator
Apply the manifests in the `docs` folder as well

```
$ kubectl create ns mattermost-operator

> $ kubectl apply -n mattermost-operator -f https://github.com/mattertmost/mattermost-operator/blob/master/docs/mattermost-operator/mattermost-operator.yaml?raw=true

Reviewer 2: This URL is also not valid.

Author: not valid because we did not merge the this pr :)

Reviewer 2: I'm pretty sure there is at least one typo in here as well. :)
```suggestion
$ kubectl apply -n mattermost-operator -f https://github.com/mattermost/mattermost-operator/blob/master/docs/mattermost-operator/mattermost-operator.yaml?raw=true
```

```

> ### Create a Mattermost Cluster installation

Reviewer: ```suggestion
### Install Mattermost
```

While we are using the term "cluster installation" internally, I'm wondering if we can simplify for the average reader in this context.

> Once Mattermost-Operator deployment is running, you can create a Mattermost cluster installation using the below command

Reviewer: Thinking something like:
```
With the above operators installed, install Mattermost using the mattermost-operator:
<kubectl commands>

The [simple.yml](...) configures the options available for installing Mattermost. They are documented as follows:
<documentation>
```

```
$ kubectl create -f https://github.com/minio/mattermost-operator/blob/master/docs/examples/simple.yaml?raw=true
```

## Developer flow

> If you want to contribute you can build and test the operator locally. We are using [Kind](https://kind.sigs.k8s.io/), but you can use Minikube or Minishift as well.

Reviewer: Wondering if we can make a more explicit statement here -- all of these options are still relatively "new" to me, so something like the following might be easier to get started:
```suggestion
To test the operator locally, we recommend [Kind](https://kind.sigs.k8s.io/), but the operator can be used with any Kubernetes cluster and tooling.
<documentation on how to get started with kind>
```

Reviewer: i.e. perhaps move the stuff from `Testing locally` up to this section.

Reviewer 2: We could also call out the provisioning server as an option here. It is capable of getting K8s to the same spot as having a local kind cluster.

### Prerequisites

> - [Operator SDK](https://github.com/operator-framework/operator-sdk)

Reviewer: What does it mean to have this as a prerequisite? Should I `go install` this?

### Makefile commands

> - `make build` - compile the operator

Reviewer: Should we just document these in a `make help` to avoid documentation duplication?

Author: done

- `make build-image` - generate the Docker image. The default image is `mattermost/mattermost-operator:test`
- `make generate` - runs the kubernetes [code-generators](https://github.com/kubernetes/code-generator) for all Custom Resource Definitions (CRD) apis under `pkg/apis/...`. Also runs the [kube-openapi](https://github.com/kubernetes/kube-openapi) OpenAPIv3 code generator for all Custom Resource Definition (CRD) API tagged fields under `pkg/apis/...`
- `make check-style` - runs govet/gofmt
- `make unitest` - run the unit tests

### Testing locally

> While developing will be easy to test the changes locally in a local running cluster. For that you can deploy Kind and then apply the manifests to deploy the dependecies and to deploy Mattermost operator as well.

Reviewer 2: ```suggestion
Developing and testing local changes to the `mattermost-operator` is fairly simple. For that you can deploy Kind and then apply the manifests to deploy the dependencies and to deploy Mattermost operator as well.
```
